src/AdminConsole/Pages/Organization/Admins.cshtml.cs

```diff
@@ -92,28 +94,39 @@ public async Task<IActionResult> OnPostDelete(string userId)
     public async Task<IActionResult> OnPostInvite(InviteForm form)
     {
         CanInviteAdmin = await _dataService.CanInviteAdminAsync();
+
         if (CanInviteAdmin is false)
         {
             ModelState.AddModelError("error", "You need to upgrade to a paid organization to invite more admins.");
             return await OnGet();
         }
 
+        Models.Organization org = await _dataService.GetOrganizationAsync();
+        var existingInvites = await _invitationService.GetInvitesAsync(org.Id);
+
+        if (existingInvites.Count == 10)
+        {
+            ModelState.AddModelError("error", "You can only have 10 pending invites at a time.");
+            return await OnGet();
+        }
+
+        if (existingInvites.Any(x => x.ToEmail == form.Email))
```

Comment:
Note that it might not catch all duplicates, including emails that use `+` or those written in a different case. If this is something that people may abuse, then maybe we need a more resilient check?

Reply:
good observation, in this case we just want a simple check. We should probably to a case-insensitive comparison.